### PR TITLE
chore(docs): Improve adherence to DTCG format module

### PR DESCRIPTION
**Pull request by VincentSmedinga**
# Improve adherence to DTCG format module

## Summary

This branch improves adherence to the [W3C Design Tokens Community Group (DTCG)](https://design-tokens.github.io/community-group/format/) specification across all token files, and upgrades the `DesignTokensTable` Storybook component to support the richer token metadata.

### Token changes (77 files, ~2700 line diff)

- **Add `$type` to all previously untyped tokens** — every token now declares its type (`color`, `dimension`, `fontFamily`, `shadow`, etc.).
- **Use group-level `$type` inheritance** — where an entire group shares a type, it's declared once on the group instead of on every token, per the DTCG spec.
- **Convert `fontFamily` values to arrays** — `"Arial, sans-serif"` becomes `["Arial", "sans-serif"]`, matching DTCG's required format.
A custom Style Dictionary transform (`dtcg-font-family.js`) converts them back to comma-separated CSS at build time.
- **Use DTCG `number` type for `lineHeight` tokens** — these are unitless ratios, not dimensions.
- **Remove non-compliant `$type: dimension`** from tokens whose values aren't valid DTCG dimensions (e.g. `cursor`, `borderStyle`).
- **Add `$description` to brand and common tokens** — human-readable documentation embedded in the token files.

### DesignTokensTable component changes

- **Group-level `$type` inheritance** — the table resolver now walks ancestor groups to inherit `$type`, matching the DTCG spec.
- **`path` and `exclude` props** — narrow which subtree to display and omit specific keys.
- **Description column** — opt-in via `showDescriptions`, populated from `$description`.
- **Deprecation warnings** — tokens with `$deprecated` show a warning icon with tooltip.
- **`AspectRatioSample`** — new visual example component for aspect ratio tokens.
- **Negative border widths** — `BorderSample` now shows absolute width.
- **Extract helpers into separate files with unit tests** — types, type guards (`isTokenValue`, `isDimensionValue`), formatters (`formatSubValue`, `formatShadowValue`), and `flattenTokens` each live in their own module with dedicated tests (40 tests total).

### Other changes

Two already-merged breaking changes are included in the branch history (`Icon` svg prop, `Grid`/`Page` compact mode) — these were merged into `main` before this branch diverged or were cherry-picked.

## Review approach

1. **Start with the token format changes** — pick 2-3 representative token files (e.g. `color.tokens.json`, `typography.tokens.json`, `accordion.tokens.json`) and verify the pattern: group-level `$type`, array `fontFamily`, `$description` fields.
Skip reviewing all 77 token files line-by-line; they follow the same mechanical pattern.
2. **Review `flattenTokens.ts`** — this is the core logic.
The test file (`flattenTokens.test.ts`) doubles as a specification of every edge case (inheritance, overrides, extensions, shadows, arrays).
3. **Review the component changes** — `DesignTokensTable.tsx` (path walking, exclude filtering) and `DesignTokensTableRow.tsx` (deprecation icon, new sample types).
4. **Skim the remaining helpers** — `formatShadowValue.ts`, `formatSubValue.ts`, and the two type guards are small and well-tested.
5. **Check `dtcg-font-family.js`** — the Style Dictionary transform that converts array fontFamily values back to CSS.

### CODE REVIEW

File: storybook/src/_components/BorderSample/BorderSample.tsx [hunk 1]
```diff
@@ -18,9 +18,10 @@ type BorderSampleProps = {
 } & HTMLAttributes<HTMLDivElement>
 
 export const BorderSample = ({ className, lineStyle, style, width, ...restProps }: BorderSampleProps) => {
-  if (width?.startsWith('-')) {
-    return null
-  }
+  if (!width) return null
+
+  // Negative border tokens represent inset box-shadows; show their absolute width in the sample.
+  const positiveWidth = width.startsWith('-') ? width.substring(1) : width
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/src/_components/BorderSample/BorderSample.tsx]: Fixed.

File: storybook/src/_components/DesignTokensTable/DesignTokensTable.tsx [hunk 2]
```diff
@@ -126,8 +28,38 @@ type DesignTokensTableRootProps = {
...
+    if (typeof subtree['$type'] === 'string') {
+      inheritedType = subtree['$type']
+    }
+
+    const next = subtree[segment]
+
+    if (typeof next === 'object' && next !== null && !Array.isArray(next)) {
+      subtree = next as Tokens
+    }
+  }
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/src/_components/DesignTokensTable/DesignTokensTable.tsx:62]: Fixed.

File: packages-proprietary/tokens/style-dictionary/transforms/dtcg-font-family.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * "'Amsterdam Sans', Arial, sans-serif"
+ * @see {@link https://www.designtokens.org/TR/2025.10/format/#font-family} DTCG fontFamily spec
+ */
+export const dtcgFontFamily = {
+  filter: (token) => Array.isArray(token.$value ?? token.value),
+  name: 'dtcg/fontFamily',
+  transform: (token) => {
+    const value = token.$value ?? token.value
+    return value.map((name) => (name.includes(' ') ? `'${name}'` : name)).join(', ')
+  },
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages-proprietary/tokens/style-dictionary/transforms/dtcg-font-family.js:29]: Fixed.

File: storybook/src/_components/DesignTokensTable/DesignTokensTableRow.tsx [hunk 4]
```diff
@@ -19,22 +25,28 @@ type DesignTokensTableRowProps = {
...
+      {deprecated && (
+        <Icon className="_ams-design-tokens-table__deprecated-icon" svg={WarningIcon} title={deprecated} />
+      )}
     </td>
     <td>
       <Code>{value}</Code>
     </td>
+    {description !== undefined && <td>{description}</td>}
     <td>
+      {type === 'aspectRatio' && <AspectRatioSample value={value} />}
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/src/_components/DesignTokensTable/DesignTokensTableRow.tsx]: Fixed.

File: packages-proprietary/tokens/src/common/ams/inputs.tokens.json [hunk 5]
```diff
@@ -41,9 +43,11 @@
...
       "line-height": {
         "$value": "1.4",
+        "$description": "Tighter than body text (1.6) to keep inputs compact. Buttons also use this to match.",
         "$extensions": {
-          "nl.amsterdam.type": "lineHeight"
-        }
+          "nl.amsterdam.subtype": "lineHeight"
+        },
+        "$type": "number"
       },
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages-proprietary/tokens/src/common/ams/inputs.tokens.json:59]: Fixed.

File: storybook/src/_components/DesignTokensTable/DesignTokensTable.types.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    /** An Amsterdam-specific subtype that takes precedence over `$type` when resolving the token's visual example. */
+    'nl.amsterdam.subtype'?: string
+    /** An Amsterdam-specific type used as a fallback when no `$type` or subtype is set. */
+    'nl.amsterdam.type'?: string
+  }
+  /** The DTCG type of the token, e.g. `'color'`, `'dimension'`, or `'shadow'`. */
+  $type?: string
+  /** The resolved value of the token. */
+  $value: DimensionValue | ShadowValue | string | string[]
+}
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/src/_components/DesignTokensTable/DesignTokensTable.types.ts:47]: Fixed.

File: packages-proprietary/tokens/src/components/ams/grid.tokens.json [hunk 7]
```diff
@@ -2,101 +2,111 @@
   "ams": {
     "grid": {
       "column-count": {
-        "$value": "4"
+        "$value": "4",
+        "$type": "number"
       },
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages-proprietary/tokens/src/components/ams/grid.tokens.json:7]: Fixed.

File: storybook/src/_components/DesignTokensTable/flattenTokens.ts [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+      if (typeof $value === 'string') {
+        normalizedValue = $value
+      } else if (Array.isArray($value)) {
+        normalizedValue = $value.map((name) => (name.includes(' ') ? `'${name}'` : name)).join(', ')
+      } else if (isDimensionValue($value)) {
+        normalizedValue = `${$value.value}${$value.unit}`
+      } else if (type === 'shadow' && typeof $value === 'object') {
+        normalizedValue = formatShadowValue($value as ShadowValue)
+      }
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/src/_components/DesignTokensTable/flattenTokens.ts:51]: Fixed.

File: storybook/src/_components/DesignTokensTable/isDimensionValue.ts [hunk 9]
```diff
@@ -0,0 +1,10 @@
+/**
+ * @license EUPL-1.2+
+ * Copyright Gemeente Amsterdam
+ */
+
+import type { DimensionValue } from './DesignTokensTable.types'
+
+/** Type guard that checks whether a value is a dimension object (i.e. has `value` and `unit` keys). */
+export const isDimensionValue = (value: unknown): value is DimensionValue =>
+  typeof value === 'object' && value !== null && 'value' in value && 'unit' in value
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/src/_components/DesignTokensTable/isDimensionValue.ts]: Fixed.

File: packages-proprietary/tokens/src/brand/ams/typography.tokens.json [hunk 10]
```diff
@@ -22,8 +29,9 @@
         "line-height": {
           "$value": "1.6",
           "$extensions": {
-            "nl.amsterdam.type": "lineHeight"
-          }
+            "nl.amsterdam.subtype": "lineHeight"
+          },
+          "$type": "number"
         },
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages-proprietary/tokens/src/brand/ams/typography.tokens.json:35]: Fixed.

File: packages-proprietary/tokens/src/components/ams/link.tokens.json [hunk 11]
```diff
@@ -22,8 +22,9 @@
       "line-height": {
         "$value": "inherit",
         "$extensions": {
-          "nl.amsterdam.type": "lineHeight"
-        }
+          "nl.amsterdam.subtype": "lineHeight"
+        },
+        "$type": "number"
       },
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages-proprietary/tokens/src/components/ams/link.tokens.json:24]: Fixed.

File: packages-proprietary/tokens/src/components/ams/unordered-list.tokens.json [hunk 12]
```diff
@@ -94,16 +95,16 @@
         },
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages-proprietary/tokens/src/components/ams/unordered-list.tokens.json:107]: Fixed.

File: storybook/src/_components/BorderSample/BorderSample.tsx [hunk 13]
```diff
@@ -33,7 +34,7 @@ export const BorderSample = ({ className, lineStyle, style, width, ...restProps
           lineStyle !== undefined
             ? formatTokenValue<CSSProperties['borderInlineStartStyle']>(lineStyle)
             : style?.borderInlineStartStyle,
-        borderInlineStartWidth: width !== undefined ? formatTokenValue(width) : style?.borderInlineStartWidth,
+        borderInlineStartWidth: positiveWidth ? formatTokenValue(positiveWidth) : undefined,
       }}
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/src/_components/BorderSample/BorderSample.tsx:38]: Fixed.

File: storybook/src/_components/AspectRatioSample/AspectRatioSample.tsx [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+  /** An aspect ratio value, e.g. "16 / 9" or "1 / 1". */
+  value: string
+} & HTMLAttributes<HTMLDivElement>
+
+export const AspectRatioSample = ({ className, style, value, ...restProps }: AspectRatioSampleProps) => (
+  <div
+    {...restProps}
+    className={clsx('_ams-aspect-ratio-sample', 'sb-unstyled', className)}
+    style={{ ...style, aspectRatio: value }}
+  />
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/src/_components/AspectRatioSample/AspectRatioSample.tsx:23]: Fixed.

File: packages-proprietary/tokens/src/components/ams/dialog.tokens.json [hunk 15]
```diff
@@ -25,22 +28,27 @@
         }
       },
       "box-shadow": {
-        "$value": "initial"
+        "$value": "initial",
+        "$type": "shadow"
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages-proprietary/tokens/src/components/ams/dialog.tokens.json]: Fixed.

File: packages-proprietary/tokens/src/components/ams/grid.tokens.json [hunk 16]
```diff
@@ -2,101 +2,111 @@
...
+            "nl.amsterdam.type": "dimension"
           }
         }
       },
       "cell": {
         "background-color": {
-          "$value": "initial"
+          "$value": "initial",
+          "$type": "color"
         },
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages-proprietary/tokens/src/components/ams/grid.tokens.json:100]: Fixed.

File: packages-proprietary/tokens/src/components/ams/page.tokens.json [hunk 17]
```diff
@@ -17,19 +17,18 @@
...
             "$value": "{ams.grid.padding-inline}",
-            "$type": "dimension",
             "$extensions": {
               "nl.amsterdam.hint": "Must be the Grid inline padding, to make sure Skip Link and Grid line up.",
               "nl.amsterdam.subtype": "space"
             }
           }
-        }
+        },
+        "$type": "dimension"
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages-proprietary/tokens/src/components/ams/page.tokens.json]: Good find, fixed.

File: storybook/src/_components/DesignTokensTable/DesignTokensTableRow.tsx [hunk 18]
```diff
@@ -11,37 +15,61 @@ import { SpaceSample } from '../SpaceSample/SpaceSample'
...
+}: DesignTokensTableRowProps) => (
   <tr>
     <td>
       <Code>var({name})</Code>
+      {deprecated && (
+        <Icon
+          className="_ams-design-tokens-table__deprecated-icon"
+          svg={WarningIcon}
+          title={`Deprecated. ${deprecated}`}
+        />
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/src/_components/DesignTokensTable/DesignTokensTableRow.tsx]: Addressed.

File: packages-proprietary/tokens/src/components/ams/accordion.tokens.json [hunk 19]
```diff
@@ -50,6 +53,22 @@
             "$type": "color"
           }
         }
+      },
+      "panel": {
+        "padding-block": {
+          "$deprecated": "Will be removed in release 4.0.0.",
+          "$value": "0",
+          "$extensions": {
+            "nl.amsterdam.type": "dimension"
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply packages-proprietary/tokens/src/components/ams/accordion.tokens.json]: Deleted them – they should have been already.

File: storybook/src/_components/DesignTokensTable/flattenTokens.ts [hunk 20]
```diff
@@ -0,0 +1,72 @@
...
+
+      return [
+        {
+          deprecated: $deprecated,
+          description: $description,
+          path: `--${currentPath.join('-')}`,
+          type,
+          value: normalizedValue,
```
[the remark this answers is not in the document]
Comment (VincentSmedinga) [reply storybook/src/_components/DesignTokensTable/flattenTokens.ts:62]: Fixed.
